utils/src/snapshots.ts

```diff
@@ -2,12 +2,14 @@ import { basename, dirname, extname, join } from 'node:path';
 import { snapshot } from 'node:test';
 
 snapshot.setResolveSnapshotPath(generateSnapshotPath);
+
+type SnapshotPath = Parameters<Parameters<typeof snapshot.setResolveSnapshotPath>[0]>[0];
+
 /**
  * @param {string} testFilePath `'/tmp/foo.test.js'`
  * @returns `'/tmp/foo.test.snap.cjs'`
- * @type {Parameters<snapshot.setResolveSnapshotPath>[0]}
  */
-function generateSnapshotPath(testFilePath) {
+function generateSnapshotPath(testFilePath: SnapshotPath) {
```

Comment:
but actually, we can get that more easily. not sure why i didn't do this before:

```suggestion
const generateSnapshotPath: Parameters<typeof snapshot.setResolveSnapshotPath>[0] = (testFilePath: SnapshotPath) => {
```

There may be an explicit type for that available, like `SnapshotPathResolver` that we can use instead of `Parameters<…>`.

Happy to leave that to a follow-up.